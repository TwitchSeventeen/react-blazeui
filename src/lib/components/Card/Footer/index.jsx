import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Footer(props) {
  const {
    children, block, className, dataName, ...other
  } = props;
  const classes = classNames('c-card__footer', { 'c-card__footer--block': block }, className);
  return (
    <footer
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </footer>
  );
}

Footer.propTypes = {
  /**
   * A block footer with no padding
   */
  block: PropTypes.bool,
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Footer.defaultProps = {
  block: false,
  children: null,
  className: null,
  dataName: 'card-footer',
};

export default Footer;
