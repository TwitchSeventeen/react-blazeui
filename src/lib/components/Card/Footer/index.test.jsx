import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Footer from './index';

describe('Card.Footer', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Footer).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Footer>
        lorem
      </Footer>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly as a block', () => {
    const { asFragment } = render(
      <Footer block>
        lorem
      </Footer>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
