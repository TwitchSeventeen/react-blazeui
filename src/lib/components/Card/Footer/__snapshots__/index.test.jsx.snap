// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.Footer renders correctly 1`] = `
<DocumentFragment>
  <footer
    class="c-card__footer"
    data-name="card-footer"
  >
    lorem
  </footer>
</DocumentFragment>
`;

exports[`Card.Footer renders correctly as a block 1`] = `
<DocumentFragment>
  <footer
    class="c-card__footer c-card__footer--block"
    data-name="card-footer"
  >
    lorem
  </footer>
</DocumentFragment>
`;
